File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/services/HouseServiceI.java
package com.chaincuerealestate.chaincuerealestate.services;

import com.chaincuerealestate.chaincuerealestate.domains.House;

import java.util.List;

public interface HouseServiceI {
    House save(House.HouseTypes houseTypes);

    House findById(String id);

    List<House> findAll();
}

File: kotlin/chaincue-real-estate-reactive/src/main/kotlin/com/chaincuerealestate/chaincuerealestatereactive/services/HouseServiceI.kt
package com.chaincuerealestate.chaincuerealestatereactive.services

import com.chaincuerealestate.chaincuerealestatereactive.domains.House
import kotlinx.coroutines.flow.Flow

interface HouseServiceI {
    suspend fun save(houseTypes: House.HouseTypes): House
    suspend fun findByIdWithRelations(id: String): House
    suspend fun findAll(): Flow<House>
}

